src/oc/storage/db/common.clj

```diff
@@ -157,4 +157,45 @@
             (r/run query conn)
             (if (= (type query) rethinkdb.net.Cursor)
               (seq query)
-              query)))))
\ No newline at end of file
+              query)))))
+
+(defn update-poll-vote
+  "
+  Atomic update of poll vote to avoid race conditions while multiple
+  users are voting together.
+  "
+  [conn table-name entry-uuid poll-uuid reply-id user-id vote?]
+  {:pre [(db-common/conn? conn)
+         (db-common/s-or-k? table-name)]}
+  (let [ts (db-common/current-timestamp)
+        set-operation (if vote? r/set-insert r/set-difference)
```

Comment:
not so clear to someone reading this (like me) when `vote?` would be false and what that means to the fn.